From: Arnd Bergmann <arnd@arndb.de>
Subject: fs/proc/task_mmu: hide unused pagemap_scan_backout_range() function
Date: Wed, 27 Sep 2023 08:02:23 +0200

This function has two callers, each of which are hidden behind an #ifdef:

fs/proc/task_mmu.c:2022:13: error: 'pagemap_scan_backout_range' defined but not used [-Werror=unused-function]

Add another conditional to check for both options to avoid the
unused-function warning.

Link: https://lkml.kernel.org/r/20230927060257.2975412-1-arnd@kernel.org
Fixes: 93538f467c0f6 ("fs/proc/task_mmu: implement IOCTL to get and optionally clear info about PTEs")
Signed-off-by: Arnd Bergmann <arnd@arndb.de>
Cc: Muhammad Usama Anjum <usama.anjum@collabora.com>
Cc: Ryan Roberts <ryan.roberts@arm.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 fs/proc/task_mmu.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/fs/proc/task_mmu.c~fs-proc-task_mmu-implement-ioctl-to-get-and-optionally-clear-info-about-ptes-fix-3
+++ a/fs/proc/task_mmu.c
@@ -1941,6 +1941,7 @@ static void make_uffd_wp_huge_pte(struct
 				make_pte_marker(PTE_MARKER_UFFD_WP), psize);
 }
 
+#if defined(CONFIG_TRANSPARENT_HUGEPAGE) || defined(CONFIG_HUGETLB_PAGE)
 static void pagemap_scan_backout_range(struct pagemap_scan_private *p,
 				       unsigned long addr, unsigned long end)
 {
@@ -1966,6 +1967,7 @@ static bool pagemap_scan_is_interesting_
 
 	return true;
 }
+#endif
 
 static bool pagemap_scan_is_interesting_vma(unsigned long categories,
 					    const struct pagemap_scan_private *p)
_
